### ISSUE TITLE
Content store: Elements are too cramped

### ISSUE THREAD

**Issue by ClobberXD** [minetest/minetest#7385]
##### Issue type

 - Bug report / Suggestion
 
##### Minetest version

```
Minetest 0.5.0-dev-75aa41c (Linux)
Using Irrlicht 1.8.3
BUILD_TYPE=Release
RUN_IN_PLACE=1
USE_GETTEXT=0
USE_SOUND=1
USE_CURL=1
USE_FREETYPE=1
USE_LUAJIT=1
STATIC_SHAREDIR="."
```

##### Summary

![image](https://user-images.githubusercontent.com/36130650/40650136-2fc230c2-6350-11e8-9e3a-5976e1f91d0c.png)

Either the number of packages per page could be reduced to 4, or the size of the browser formspec could be increased (recommended).

A character limit could be set to the package's short description to prevent huge "short" descriptions, and subsequently eliminate the need for scroll-bars.

EDIT: Another screenshot to emphasize this issue better.
![image](https://user-images.githubusercontent.com/36130650/40901375-96ef43f6-67ed-11e8-8e43-04e2d8c663f3.png)

##### Steps to reproduce

Navigate the online content browser until you see a package with a massive "short" description.

**Comment by numberZero** [comment]
1. Form is small, there is more space available (at least on PC screen).
2. No free space between items is bad, especially as there *is* some free space between the name and the description.
3. Adding horizontal lines isn’t the best idea, they will probably only increase the clutter.
4. It refreshes at every image load, preventing from using any input field.
5. The View and Install buttons are huge for such a terse list. The description font could be smaller **a bit** as well.

**Comment by ClobberXD** [comment]
Will work on this when #7497 has been merged.

**Comment by ClobberXD** [comment]
What if no short description is displayed on the main browser screen? Instead, the install and view buttons can be placed below the title, and the view button will display the long description as usual. Will post a screenshot to demonstrate this within a couple of days.

**Comment by paramat** [comment]
@rubenwardy

**Comment by paramat** [comment]
I agree with numberZero's points 1 2 3 5.

**Comment by Fixer-007** [comment]
Scrollbars :S

**Comment by ClobberXD** [comment]
Since there's a separate package dialog formspec (`"View"`) anyway, I think we can completely get rid of the short desc. in the content store formspec. Is it a good idea to expand the content store formspec horizontally?

**Comment by niansa** [comment]
I think it would be the best to just make it dynamic to the window/screen size.

**Comment by niansa** [comment]
@ClobberXD it has been merged.

**Comment by ClobberXD** [comment]
@niansa Uhh... what exactly has been merged?

**Comment by ClobberXD** [comment]
@niansa Assuming you're referring to #7497, that isn't the ideal fix to this issue, and the root causes (size and improper layout) would still remain. I'd prefer this issue to be tackled by more concrete solutions, and those solutions are yet to be decided upon.

**Comment by paramat** [comment]
This formspec does scale with game window size, so could cover more of the screen.

![screenshot from 2018-10-21 19-20-50](https://user-images.githubusercontent.com/3686677/47270701-7bca3f80-d567-11e8-9fcd-4664e114a9ac.png)

^ With the default 1024x600 game window size.

There is wasted space under the dropdown. How about shortening the search bar and moving the dropdown up beside the search bar? Or better, move the buttons at the bottom up to beside the dropdown? Then the packages can be full width.

**Comment by rubenwardy** [comment]
The dropdown should be moved to be level with the searchbar, and each entry made wider

The dropdown was there originally because that's where I was planning to put tags. I doubt tags will be in 5.0 though, so best to move it

**Comment by random-geek** [comment]
@rubenwardy @paramat What if we just reduced the page to 4 items each? There would be lots more space, and I don't think it really matters whether there are 4 or 5 items on a page.

**Comment by ClobberXD** [comment]
I did suggest that initially, but considering the sheer number of packages the content store would potentially have in the future, that doesn't seem to be a good idea.

**Comment by paramat** [comment]
I'm fine with 5 per page, the descriptions are scrollable.

**Comment by numberZero** [comment]
Scrollable short descriptions are nonsense.

**Comment by ClobberXD** [comment]
Fixed in #7816
